dotCMS/src/main/java/com/dotcms/cms/login/LoginServiceAPIImpl.java

```diff
@@ -0,0 +1,569 @@
+                    Logger.error(this.getClass(), "Login Failed: " + e1);
+                }
+            }
+            SecurityLogger.logInfo(this.getClass(), "An invalid attempt to login as "
+                            + userIdOrEmail + " from IP: " + request.getRemoteAddr());
+
+
+            Logger.error(this.getClass(), "Login Failed: " + e);
+            SecurityLogger.logInfo(this.getClass(), "An invalid attempt to login as "
+                            + userIdOrEmail + " has been made from IP: " + request.getRemoteAddr());
+        }
+
+        return false;
+    }
+
+    @Override
+    public boolean doLogin(final String userName, final String password)
+                    throws NoSuchUserException {
+
+        HttpServletRequest request =
+                        new MockSessionRequest(new MockHttpRequest("fake", "fake").request())
+                                        .request();
+        HttpServletResponse response =
+                        new MockHttpResponse(new BaseResponse().response()).response();
+
+        return _doLogin(userName, password, false, request, response, false);
+
+
+
+    }
+
+    @Override
+    public boolean passwordMatch(String password, User user) throws DotSecurityException {
+
+
+        try {
+            boolean needsToRehashPassword = false;
+            if (PasswordFactoryProxy.isUnsecurePasswordHash(user.getPassword())) {
+                // This is the legacy functionality that we will removed with
+                // the new hash library
+                boolean match = user.getPassword().equals(password) || user.getPassword()
+                                .equals(PublicEncryptionFactory.digestString(password));
+
+                // Bad credentials
+                if (!match) {
+                    throw new DotSecurityException("passwords don't match");
+                }
+
+                // Force password rehash
+                needsToRehashPassword = true;
+            } else {
+                PasswordFactoryProxy.AuthenticationStatus authStatus =
+                                PasswordFactoryProxy.authPassword(password, user.getPassword());
+
+                // Bad credentials
+                if (authStatus.equals(
+                                PasswordFactoryProxy.AuthenticationStatus.NOT_AUTHENTICATED)) {
+                    throw new DotSecurityException("passwords don't match");
+                }
+
+                // Force password rehash
+                if (authStatus.equals(PasswordFactoryProxy.AuthenticationStatus.NEEDS_REHASH)) {
+                    needsToRehashPassword = true;
+                }
+            }
+
+            // Apply new hash to the password and update user
+            if (needsToRehashPassword) {
+                // We need to rehash password and save the new ones
+                user.setPassword(PasswordFactoryProxy.generateHash(password));
+                user.setLastLoginDate(new java.util.Date());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use the Java 8 Date and Time API instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2143)